nflow-engine/src/main/java/io/nflow/engine/internal/dao/WorkflowInstanceDao.java

```diff
@@ -668,36 +685,54 @@ public int compareTo(OptimisticLockKey other) {
     }
     conditions.add("executor_group = :executor_group");
     params.addValue("executor_group", executorInfo.getExecutorGroup());
+
+    String innerJoinMain = "";
+    String innerJoinArchive = "";
     if (query.stateVariableKey != null) {
-      sqlBuilder.append("inner join nflow_workflow_state wfs on wf.id = wfs.workflow_id and wfs.state_key = :state_key and wfs.state_value = :state_value ");
+      innerJoinMain = "inner join " + MAIN.nameOf("workflow_state") + " wfs on wf.id = wfs.workflow_id and wfs.state_key = :state_key and wfs.state_value = :state_value ";
+      innerJoinArchive = "inner join " + ARCHIVE.nameOf("workflow_state") + " wfs on wf.id = wfs.workflow_id and wfs.state_key = :state_key and wfs.state_value = :state_value ";
       conditions.add("wfs.action_id = (select max(action_id) from nflow_workflow_state where workflow_id = wf.id and state_key = :state_key)");
       params.addValue("state_key", query.stateVariableKey);
       params.addValue("state_value", query.stateVariableValue);
     }
-    String sql = sqlBuilder.append("where ").append(collectionToDelimitedString(conditions, " and ")).append(" order by id desc").toString();
-    sql = sqlVariants.limit(sql, getMaxResults(query.maxResults));
 
-    Stream<WorkflowInstance> ret = namedJdbc.query(sql, params, new WorkflowInstanceRowMapper()).stream()
-        .map(WorkflowInstance.Builder::build);
+    String whereCondition = " where " + collectionToDelimitedString(conditions, " and ") + " order by id desc";
+
```

Comment:
If you need to have empty lines in a method for readability, maybe it is too long anyway and should be splitted or some parts extracted to make it more readable?